llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -146,15 +140,14 @@ MachineTraceMetrics::getProcReleaseAtCycles(unsigned MBBNum) const {
          "getResources() must be called before getProcReleaseAtCycles()");
   unsigned PRKinds = SchedModel.getNumProcResourceKinds();
   assert((MBBNum+1) * PRKinds <= ProcReleaseAtCycles.size());
-  return ArrayRef(ProcReleaseAtCycles.data() + MBBNum * PRKinds, PRKinds);
+  return ArrayRef{ProcReleaseAtCycles.data() + MBBNum * PRKinds, PRKinds};
```

Comment:
We don't really use `ClassName{}` style constructors inside LLVM unless there is some specific reason to do so. This should either stay as `ArrayRef()` or become `{}` without mentioning the name at all.